### PR TITLE
Added events SpawnHibernate, SpawnScout

### PR DESCRIPTION
**Pull request by Inas-07**

### CODE REVIEW

File: AWO/Modules/WEE/Events/Enemy/SpawnScoutInZoneEvent.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if (ss.AreaIndex != -1)
+        {
+            if (ss.AreaIndex < 0 || ss.AreaIndex >= zone.m_areas.Count)
+            {
+                Logger.Error($"Invalid AreaIndex {ss.AreaIndex} for ZONE_{zone.Alias}");
+                return;
+            }
+        }
+
+        CoroutineDispatcher.StartCoroutine(DoSpawn(e, zone));
```
Comment (Flowaria) [inline AWO/Modules/WEE/Events/Enemy/SpawnScoutInZoneEvent.cs]: Use `CoroutineDispatcher.StartInLevelCoroutine` would be better for this case

File: AWO/Modules/WEE/Events/Enemy/SpawnScoutInZoneEvent.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    static IEnumerator DoSpawn(WEE_EventData e, LG_Zone zone)
+    {
+        var ss = e.SpawnScouts;
+        System.Random rand = ss.AreaIndex == -1 ? new() : null;
```
Comment (Flowaria) [inline AWO/Modules/WEE/Events/Enemy/SpawnScoutInZoneEvent.cs]: If set seed is not needed there is `AWO.Utils.RNG` class that have System.Random instance to work with, Could add new method for RandomRange and use that instead

**Review by Flowaria — COMMENTED** [review]
LGTM besides few nitpick

**Comment by Inas-07** [comment]
Done accordingly

**Comment by DLX9986** [comment]
@Inas-07 可以授权给我你们团队未公开的mod使用权吗

**Pull request merged**